lib/saml_idp/logout_request_builder.rb

```diff
@@ -1,26 +1,31 @@
 require 'saml_idp/logout_builder'
 module SamlIdp
   class LogoutRequestBuilder < LogoutBuilder
+    include SamlIdp::Signable
+
     attr_accessor :name_id
+    attr_accessor :reference_id
 
     def initialize(response_id, issuer_uri, saml_slo_url, name_id, algorithm)
       super(response_id, issuer_uri, saml_slo_url, algorithm)
       self.name_id = name_id
     end
 
     def build
+      req_options = {}
+      req_options[:ID] = "_#{reference_id}"
+      req_options[:Version] = "2.0"
+      req_options[:IssueInstant] = now_iso
+      req_options[:Destination] = saml_slo_url
+      req_options["xmlns:samlp"] = Saml::XML::Namespaces::PROTOCOL
+      req_options["xmlns:saml"] = Saml::XML::Namespaces::ASSERTION
+      req_options[:NotOnOrAfter] = (Time.now + 180).utc.iso8601
       builder = Builder::XmlMarkup.new
-      builder.LogoutRequest ID: response_id_string,
```

Comment:
There's no particular reason for removing the block. I had applied a monkey patch to enable IdP-initiated single logout in my service, and I submitted that change with this PR.

However, I will add the SessionIndex since it seems to be necessary for SP-initiated single logout.